packages/ef-runtime-client/src/EFRuntime/EFRuntime.ts

```diff
@@ -0,0 +1,87 @@
+import { IComponentRegistry } from "../ComponentRegistry";
+import { ModuleLoader } from "../ModuleLoader";
+import { StylingHandler } from "../StylingHandler";
+
+export interface IRuntimeDependencies {
+  componentRegistry: IComponentRegistry;
+  moduleLoader: ModuleLoader;
+  stylingHandler: StylingHandler;
+}
+
+export class EFRuntime {
+  private registry: IComponentRegistry;
+  private moduleLoader: ModuleLoader;
+  private stylingHandler: StylingHandler;
+
+  constructor({
+    componentRegistry,
+    moduleLoader,
+    stylingHandler,
+  }: IRuntimeDependencies) {
+    this.registry = componentRegistry;
+    this.moduleLoader = moduleLoader;
+    this.stylingHandler = stylingHandler;
+  }
+
+  private validateOptions(options: {
+    systemCode?: string;
+    overrides?: { [propName: string]: string };
+  }): void {
+    if (!options.systemCode) {
+      throw new Error("Must provide a systemCode option");
+    }
+  }
+
+  async init(
+    options: {
+      systemCode?: string;
```

Comment:
Is the systemCode really optional? If the page doesn't provide systemCode what will the registry provide?